src/main/java/org/zalando/nakadi/controller/CursorOperationsController.java

```diff
@@ -54,21 +56,29 @@
     private final CursorConverter cursorConverter;
     private final CursorOperationsService cursorOperationsService;
     private final EventTypeRepository eventTypeRepository;
+    private final AuthorizationService authorizationService;
 
     @Autowired
     public CursorOperationsController(final CursorOperationsService cursorOperationsService,
-                             final CursorConverter cursorConverter, final EventTypeRepository eventTypeRepository) {
+                                      final CursorConverter cursorConverter,
+                                      final EventTypeRepository eventTypeRepository,
+                                      final AuthorizationService authorizationService) {
         this.cursorOperationsService = cursorOperationsService;
         this.cursorConverter = cursorConverter;
         this.eventTypeRepository = eventTypeRepository;
+        this.authorizationService = authorizationService;
     }
 
     @RequestMapping(path = "/event-types/{eventTypeName}/cursor-distances", method = RequestMethod.POST)
     public ResponseEntity<?> getDistance(@PathVariable("eventTypeName") final String eventTypeName,
                                          @Valid @RequestBody final ValidListWrapper<CursorDistance> queries,
-                                         final Client client) {
+                                         final Client client) throws InternalNakadiException, NoSuchEventTypeException {
+        // TODO: remove once new authorization is in place
         checkReadScopes(eventTypeName, client);
 
+        final EventType eventType = eventTypeRepository.findByName(eventTypeName);
+        authorizeStreamRead(authorizationService, client, eventType);
```

Comment:
I don't know. I gave it some thought and it seems wrong to have to edit every single method signature to include a client. It feels like coupling the execution of an operation with checking if that operation is authorized. It might be that I already have a fixed idea in mind because when I implemented this kind of feature before, it would usually be implemented using some aspects or something outside of the actual operation.